cmd/kube-bench-summarizer/summarizer/summarizer.go

```diff
@@ -0,0 +1,200 @@
+		if err != nil {
+			logrus.Errorf("error reading file %+v: %v", fileName, err)
+			return err
+		}
+
+		results := &kb.Controls{}
+		err = json.Unmarshal(contents, results)
+		if err != nil {
+			logrus.Errorf("error unmarshalling: %v", err)
+			return err
+		}
+		logrus.Debugf("results: %+v", results)
+
+		for _, group := range results.Groups {
+			for _, check := range group.Checks {
+				logrus.Infof("id: %v %v", check.ID, check.State)
+				if check.Scored && check.State != kb.PASS {
+					logrus.Infof("check: %+v", check)
+					g := s.groupsMap[group.ID]
+					if g == nil {
+						logrus.Infof("group not found in map: %v", group.ID)
+						g = &GroupWrapper{
+							ID:     group.ID,
+							Text:   group.Text,
+							Checks: []*CheckWrapper{},
+						}
+						s.groupsMap[group.ID] = g
+						s.summarizedReport.Groups = append(s.summarizedReport.Groups, g)
+					}
+
+					c := s.checksMaps[check.ID]
+					if c == nil {
+						logrus.Infof("check not found in map: %v", check.ID)
+						c = &CheckWrapper{
+							ID:          check.ID,
+							Text:        check.Text,
+							Remediation: check.Remediation,
+							TestInfo:    check.TestInfo,
+							State:       check.State,
+							Hosts:       []string{},
+						}
+						s.checksMaps[check.ID] = c
+						g.Checks = append(g.Checks, c)
+
+						switch check.State {
+						case kb.FAIL:
+							g.Fail++
+							s.summarizedReport.Fail++
+						case kb.WARN:
+							g.Warn++
+							s.summarizedReport.Warn++
+						case kb.INFO:
+							g.Info++
+							s.summarizedReport.Info++
+						}
+					}
+					c.Hosts = append(c.Hosts, hostname)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (s *Summarizer) save() error {
+	data, err := json.MarshalIndent(s.summarizedReport, "", " ")
+	if err != nil {
+		logrus.Errorf("error marshaling: %v", err)
+		return err
+	}
+
+	outputFilePath := fmt.Sprintf("%s/%s", s.OutputDirectory, DEFAULT_OUTPUT_FILE_NAME)
+	err = ioutil.WriteFile(outputFilePath, data, 0644)
+	if err != nil {
+		logrus.Errorf("error writing report file: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+func (s *Summarizer) Summarize() error {
+	logrus.Infof("summarize")
+	logrus.Debugf("inputDir: %v", s.InputDirectory)
+
+	// Walk through the host folders
+	hostsDir, err := ioutil.ReadDir(s.InputDirectory)
+	if err != nil {
+		logrus.Errorf("error listing directory: %v", err)
+		return err
+	}
+
+	for _, hostDir := range hostsDir {
+		if !hostDir.IsDir() {
+			continue
+		}
+		hostname := hostDir.Name()
+		logrus.Debugf("hostDir: %v", hostname)
+
+		if err := s.summarizeForHost(hostname); err != nil {
+			logrus.Errorf("error summarizeForHost: %v", hostname)
+		}
+	}
+
+	return s.save()
+}
+
+func (s *Summarizer) printSummaryReport() error {
+	bytes, err := json.Marshal(s.summarizedReport)
+	if err != nil {
+		logrus.Errorf("error marshalling summary report: %v", err)
+		return err
+	}
+
+	txt := string(bytes)
+	logrus.Infof("txt: %+v", txt)
```

Comment:
do we intent to print it in the log? or was it more for debug purpose (in this case , we should reduce the log level to debug)